### PR TITLE
feat: validation schema updates

### LINKED ISSUE #240: Add validation report related entities to DB schema
Create DB entities to support the validation pipeline and the API

### PR DESCRIPTION
**Pull request by cka-y**
Closes #240 

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: liquibase/changes/feat_240.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-- 1. Add Table ValidationReport
+CREATE TABLE ValidationReport (
+    id VARCHAR(255) PRIMARY KEY,
+    dataset_id VARCHAR(255) REFERENCES GTFSDataset(id),
+    version VARCHAR(255)
```
Comment (davidgamez) [inline liquibase/changes/feat_240.sql:5]: suggestion:
```suggestion
    validator_version VARCHAR(255)
```

File: liquibase/changes/feat_240.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+CREATE TYPE severity_type AS ENUM ('ERROR', 'WARNING', 'INFO');
+
+-- 4. Add Table Notice
+CREATE TABLE Notice (
+    dataset_id VARCHAR(255) REFERENCES GTFSDataset(id),
+    validation_report_id VARCHAR(255) REFERENCES ValidationReport(id),
+    notice_code VARCHAR(255),
+    severity severity_type,
+    PRIMARY KEY (dataset_id, validation_report_id, notice_code)
+);
```
Comment (davidgamez) [inline liquibase/changes/feat_240.sql:22]: Please add the total notices field:
```suggestion
CREATE TABLE Notice (
    dataset_id VARCHAR(255) REFERENCES GTFSDataset(id),
    validation_report_id VARCHAR(255) REFERENCES ValidationReport(id),
    notice_code VARCHAR(255),
    severity severity_type,
    total_notices INT NOT NULL CHECK (total_notices > 0),
    PRIMARY KEY (dataset_id, validation_report_id, notice_code)
);
```

Comment (cka-y) [reply liquibase/changes/feat_240.sql:24]: should it be within the `Notice` entity?

Comment (davidgamez) [reply liquibase/changes/feat_240.sql:24]: This field represents the total number of notices of that notice code in the report. That's why it should be in the Notice entity. To illustrate, find below a snipped of a  report:
```
{
  "summary": {
.....
 },
  "notices": [
    {
      "code": "block_trips_with_overlapping_stop_times",
      "severity": "ERROR",
      "totalNotices": 470,  <=== This is the field
      "sampleNotices": [.....]
}]
}
```

**Review by davidgamez — APPROVED** [review]

**Pull request merged**